src/parser/transform/transform_expression.cpp

```diff
@@ -579,8 +579,15 @@ std::string Transformer::transformPropertyKeyName(CypherParser::OC_PropertyKeyNa
 std::unique_ptr<ParsedExpression> Transformer::transformIntegerLiteral(
     CypherParser::OC_IntegerLiteralContext& ctx) {
     auto text = ctx.DecimalInteger()->getText();
-    auto value =
-        std::make_unique<Value>(function::castStringToNum<int64_t>(text.c_str(), text.length()));
+    function::IntegerCastData<int64_t> data{};
+    auto len = text.length();
+    if (tryIntegerCast<function::IntegerCastData<int64_t>, true>(text.c_str(), len, data)) {
```

Comment:
This line doesn't compile on mac, guess you need to cast len to write `(uint64_t&)len`